cmd/jaeger/internal/exporters/storageexporter/exporter.go

```diff
@@ -30,14 +33,23 @@ func newExporter(config *Config, otel component.TelemetrySettings) *storageExpor
 	}
 }
 
-func (exp *storageExporter) start(_ context.Context, host component.Host) error {
+func (exp *storageExporter) start(ctx context.Context, host component.Host) error {
 	f, err := jaegerstorage.GetStorageFactory(exp.config.TraceStorage, host)
 	if err != nil {
 		return fmt.Errorf("cannot find storage factory: %w", err)
 	}
 
-	if exp.spanWriter, err = f.CreateSpanWriter(); err != nil {
-		return fmt.Errorf("cannot create span writer: %w", err)
+	switch t := f.(type) {
+	case *ch.Factory:
```

Comment:
it seems you're doing this too late, I think `createTracesExporter` would be called before `start`